### PR TITLE
beagle 3.0.2 (new formula)

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

c.f. https://github.com/Homebrew/homebrew-core/pull/28899 https://github.com/beast-dev/beast-mcmc/issues/994

### CODE REVIEW

File: Formula/beagle.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Beagle < Formula
+  desc "Evaluate the likelihood of sequence evolution on trees"
+  homepage "https://github.com/beagle-dev/beagle-lib"
+  url "https://github.com/beagle-dev/beagle-lib/archive/v3.0.2.tar.gz"
+  sha256 "59a7081b61ead0a5738e813c6fcfb614d2c5bb49c29b28609de8e2b51bea3ec0"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "doxygen" => :build
+  depends_on :java => :build
```
Comment (ilovezfs) [inline Formula/beagle.rb]: really only build-time?

Comment (jonchang) [reply Formula/beagle.rb]: Yup, `beagle` builds Java bindings to its native library, but not all downstream apps need it. See [mrbayes](https://github.com/brewsci/homebrew-bio/blob/master/Formula/mrbayes.rb). Haven't figured out a way to disable it and `beast` needs the Java bindings anyway which is why we're migrating it in the first place...

**Comment by jonchang** [comment]
N.B. `brew audit --new-formula beast` fails the new-ish notability change. Since `beast` is notable enough and we're only importing `beagle` because of `beast` I'm hoping this is OK.

File: Formula/beagle.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/beagle.rb]: maybe `=> [:build, :test]` then and add a test for the Java bindings? (see Octave)

**Comment by ilovezfs** [comment]
>I'm hoping this is OK.

yes it's fine

**Comment by ilovezfs** [comment]
Thanks for the 🆕 formula @jonchang! 🔬

**Pull request closed without merging**